composer/loggers/slack_logger.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Callable, Dict, Optional
+
+from slack_sdk import WebhookClient
+
+from composer.loggers.logger_destination import LoggerDestination
+
+
+class SlackLogger(LoggerDestination):
+    """Log metrics to slack, using `Slack Webhook URL <https://api.slack.com/messaging/webhooks>`
+    
+    .. note::
+        See example instantiating of SlackLogger in `examples/slack_log_metrics.py`
+        Webhook URL will be passed as an environment variable.
+        Formatter function can be defined at callsite or default formatter function will be applied if not provided.
+         
+    Args:
+        webhook_url (str): Slack Webhook URL
+        log_metrics_formatter_func ((...) -> Any | None): A formatter function that returns list of blocks to be sent to slack
+    """
+
+    def __init__(
+        self,
+        webhook_url: str,
+        log_metrics_formatter_func: Optional[Callable] = None,
+    ) -> None:
+        self.webhook_url = webhook_url
+        self.logged_metrics: Dict[str, float] = {}
+        self.log_metrics_formatter_func = log_metrics_formatter_func if log_metrics_formatter_func else self._default_log_bold_key_normal_value_pair_with_header
+
+    # Rich message layouts can be created using Slack Blocks Kit.
+    # See documentation here: https://api.slack.com/messaging/composing/layouts
+    def _log_to_slack(
```

Comment:
I think this function should either not be on the class, and just accept all the args (i.e. not use `self.webhook_url`), or use everything via `self` (i.e. `self.client`, `self.log_metrics_formatter_func`, `self.webhook_url`).

Reply:
I updated to use this function for metrics, traces and hparams, so I needed to parametrize it